File: src/Entity/Session.php
<?php

namespace App\Entity;

use App\Repository\SessionRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: SessionRepository::class)]
class Session
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[Assert\NotBlank(message: "L'heure de début est obligatoire.")]
    #[Assert\Type(type: "\DateTimeInterface", message: "Veuillez entrer une date et une heure valides.")]
    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $startHour = null;

    #[Assert\NotBlank(message: "L'heure de fin est obligatoire.")]
    #[Assert\Type(type: "\DateTimeInterface", message: "Veuillez entrer une date et une heure valides.")]
    #[Assert\GreaterThan(
        propertyPath: "startHour",
        message: "L'heure de fin doit être postérieure à l'heure de début."
    )]
    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $endHour = null;

    // #[ORM\Column]
    // private ?bool $sessionPresence = null;

    #[ORM\Column(length: 255)]
    private ?string $typeSession = null;

    #[ORM\ManyToOne(inversedBy: 'sessions')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Event $event = null;

    /**
     * @var Collection<int, Attendance>
     */
    #[ORM\OneToMany(targetEntity: Attendance::class, mappedBy: 'session')]
    private Collection $attendances;


    #[ORM\Column(length: 255, nullable: true)]
    private ?string $objective = null;

    #[ORM\Column]
    private ?bool $meetingStarted = null;



    public function __construct()
    {
        $this->attendances = new ArrayCollection();
        $this->startHour = new \DateTime();
        $this->endHour = new \DateTime();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getStartHour(): ?\DateTimeInterface
    {
        return $this->startHour;
    }

    public function setStartHour(\DateTimeInterface $startHour): static
    {
        $this->startHour = $startHour;

        return $this;
    }

    public function getEndHour(): ?\DateTimeInterface
    {
        return $this->endHour;
    }

    public function setEndHour(\DateTimeInterface $endHour): static
    {
        $this->endHour = $endHour;

        return $this;
    }

    // public function isSessionPresence(): ?bool
    // {
    //     return $this->sessionPresence;
    // }

    // public function setSessionPresence(bool $sessionPresence): static
    // {
    //     $this->sessionPresence = $sessionPresence;

    //     return $this;
    // }

    public function getTypeSession(): ?string
    {
        return $this->typeSession;
    }

    public function setTypeSession(string $typeSession): static
    {
        $this->typeSession = $typeSession;

        return $this;
    }

    public function getEvent(): ?Event
    {
        return $this->event;
    }

    public function setEvent(?Event $event): static
    {
        $this->event = $event;

        return $this;
    }

    /**
     * @return Collection<int, Attendance>
     */
    public function getAttendances(): Collection
    {
        return $this->attendances;
    }

    public function addAttendance(Attendance $attendance): static
    {
        if (!$this->attendances->contains($attendance)) {
            $this->attendances->add($attendance);
            $attendance->setSession($this);
        }

        return $this;
    }

    public function removeAttendance(Attendance $attendance): static
    {
        if ($this->attendances->removeElement($attendance)) {
            // set the owning side to null (unless already changed)
            if ($attendance->getSession() === $this) {
                $attendance->setSession(null);
            }
        }

        return $this;
    }

    public function getObjective(): ?string
    {
        return $this->objective;
    }

    public function setObjective(?string $objective): static
    {
        $this->objective = $objective;

        return $this;
    }

    public function isMeetingStarted(): ?bool
    {
        return $this->meetingStarted;
    }

    public function setMeetingStarted(bool $meetingStarted): static
    {
        $this->meetingStarted = $meetingStarted;

        return $this;
    }

}

File: rt/src/main/java/pi2425/swappy_javafx/entities/Session.java
package pi2425.swappy_javafx.entities;

import java.time.LocalDateTime;

public class Session {

    private int id;
    private int event_id;
    private LocalDateTime start_hour;
    private LocalDateTime end_hour;
    private String objective;
    private String type_session;
    private boolean meeting_started;
    private Event event;


    public Session() {
    }

    public Session(Event event,int id, int event_id, LocalDateTime start_hour, LocalDateTime end_hour, String objective, String type_session, boolean meeting_started) {
        this.event=event;
        this.id = id;
        this.event_id = event_id;
        this.start_hour = start_hour;
        this.end_hour = end_hour;
        this.objective = objective;
        this.type_session = type_session;
        this.meeting_started = meeting_started;
    }

    public Session(Event event,int event_id, LocalDateTime start_hour, LocalDateTime end_hour, String objective, String type_session, boolean meeting_started) {
        this.event=event;
        this.event_id = event_id;
        this.start_hour = start_hour;
        this.end_hour = end_hour;
        this.objective = objective;
        this.type_session = type_session;
        this.meeting_started = meeting_started;
    }

    public int getId() {
        return id;
    }

    public Event getEvent() {
        return event;
    }

    public void setEvent(Event event) {
        this.event = event;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setEvent_id(int event_id) {
        this.event_id = event_id;
    }

    public int getEvent_id() {
        return event_id;
    }

    public void setEventId(int event_id) {
        this.event_id = event_id;
    }

    public LocalDateTime getStart_hour() {
        return start_hour;
    }

    public void setStart_hour(LocalDateTime start_hour) {
        this.start_hour = start_hour;
    }

    public LocalDateTime getEnd_hour() {
        return end_hour;
    }

    public void setEnd_hour(LocalDateTime end_hour) {
        this.end_hour = end_hour;
    }

    public String getObjective() {
        return objective;
    }

    public void setObjective(String objective) {
        this.objective = objective;
    }

    public String getType_session() {
        return type_session;
    }

    public void setType_session(String type_session) {
        this.type_session = type_session;
    }

    public boolean isMeeting_started() {
        return meeting_started;
    }

    public void setMeeting_started(boolean meeting_started) {
        this.meeting_started = meeting_started;
    }

    @Override
    public String toString() {
        return "Session{" +
                "id=" + id +
                ", event=" + event_id +
                ", start_hour=" + start_hour +
                ", end_hour=" + end_hour +
                ", objective='" + objective + '\'' +
                ", type_session=" + type_session +
                ", meeting_started=" + meeting_started +
                '}';
    }
}
